src/lib/github/apiActions/baseApiActions.ts

```diff
@@ -0,0 +1,121 @@
+import { getAccessToken } from "@/lib/auth/getAccessToken";
+
+export type GitHubPagination = {
+  nextPage?: number;
+  firstPage?: number;
+  lastPage?: number;
+  previousPage?: number;
+};
+export type GitHubPaginatedResponse<T> = {
+  data: T;
+  pagination?: GitHubPagination;
+};
+export function BaseGithubApiActions() {
+  const baseUrl: string = "https://api.github.com";
+
+  async function get<T>(path: string): Promise<T> {
+    const bearerToken = await getBearerToken();
+    const response = await fetch(`${baseUrl}${path}`, {
+      headers: {
+        Authorization: bearerToken,
+      },
+    });
+    if (!response.ok) {
+      throw new Error(`GitHub API request failed: ${response.statusText}`);
+    }
+    const data = await response.json();
+    return data;
+  }
+
+  async function getWithPagination<T>(
+    path: string,
+    pageNumber: number = 1
+  ): Promise<GitHubPaginatedResponse<T>> {
+    const bearerToken = await getBearerToken();
+    const queryIndicator = path.includes("?") ? "&" : "?";
+    const response = await fetch(
+      `${baseUrl}${path}${queryIndicator}page=${pageNumber}`,
+      {
+        headers: {
+          Authorization: bearerToken,
+        },
+      }
+    );
+    if (!response.ok) {
+      throw new Error(`GitHub API request failed: ${response.statusText}`);
+    }
+    const linkHeader = response.headers.get("Link"); // Link gives us pagination info
+    let nextPage: number | undefined = undefined;
```

Comment:
I think you can just do `let nextPage: number | undefined;`. Not providing an initial value is the same as setting the initial value to `undefined`.